src/util/diproche/diprocheInterface.ts

```diff
@@ -0,0 +1,97 @@
+import Issue from "../../issueHandling/issue";
+import { concatOneIssueList } from "../../issueHandling/issueMapping";
+import { getAllIssues } from "../../vocabularyChecker/detectWrongSyntax";
+import { UnexpectedError } from "./Errors";
+import modes from "./predicateList.json";
+
+export type Mode = "propositionalLogic" | "firstOrderPredicateLogic" | "test";
+
+/**
+ * Since the user input most likely lacks the correct predicate,
+ * it needs to be added. For easier maintainability in case
+ * of an expanding predicate list, the corresponding predicate is
+ * added with respect to the current @param mode. This can, for
+ * example correspond to the concept of "Spielwiese". The user should
+ * be able to just stay inside a mode and not have to add this mode
+ * himself. Hence, the mode needs to be read out from the browser.
+ */
+export function addPredicate(userInput: string, mode: Mode) {
+	for (const predicate of modes.predicates) {
+		if (predicate === mode) {
+			return getPrologPredicateForMode(mode).concat("(").concat(userInput).concat(").");
+		}
+	}
+	return "Es gibt kein passendes Prädikat.";
+}
+
+export function getPrologPredicateForMode(mode: Mode) {
+	switch (mode) {
+		case "propositionalLogic": {
+			return "diproche";
+		}
+		case "firstOrderPredicateLogic": {
+			return "diproche_fo";
+		}
+		case "test": {
+			return "teste";
+		}
+		default: {
+			throw(UnexpectedError);
+		}
+	}
+}
+
+// Assuming "wrongWord" is not an allowed word, then a
```

Comment:
Examples should be described in tests.

Reply:
That was actually something I was wondering, too. As in: I know that examples should serve as tests, but I never figured, that examples should **exclusively** be tests. Here I figured a example would result in a better understanding in the code without having to look at the tests.

Reply:
Users of your code will usually look for examples on how to use your method by using "Find References" (or similar) in their IDE. This way, they will find your tests really quickly.
Ideally though, your function needs no description at all, because its name+parameter(-names+-types)+returntype already tell the whole story. A quick one-line-explanaition can clarify something that is easily misunderstood. Nobody will read a longer method-comment though, when they don't understand at this point what's happening, it's more likely they will take a look at the method-code directly.